crates/common/consensus/src/deneb/execution_payload.rs

```diff
@@ -36,3 +41,93 @@ pub struct ExecutionPayload {
+        keccak256(buf)
+    }
+
+    fn encode(&self, out: &mut dyn bytes::BufMut, parent_beacon_block_root: B256) {
+        let transactions = self
+            .transactions
+            .clone()
+            .into_iter()
+            .map(|transaction| Bytes::from(transaction.to_vec()))
+            .collect::<Vec<_>>();
+        let transactions_root = calculate_transactions_root(&transactions);
+        let withdrawals_root = calculate_withdrawals_root(&self.withdrawals);
+        alloy_rlp::Header {
+            list: true,
+            payload_length: self.rlp_payload_length(
+                parent_beacon_block_root,
+                transactions_root,
+                withdrawals_root,
+            ),
+        }
+        .encode(out);
+
+        self.parent_hash.encode(out);
+        EMPTY_UNCLE_ROOT_HASH.encode(out);
+        self.fee_recipient.encode(out);
+        self.state_root.encode(out);
+        transactions_root.encode(out);
+        self.receipts_root.encode(out);
+        self.logs_bloom.encode(out);
+        U256::ZERO.encode(out);
+        self.block_number.encode(out);
+        self.gas_limit.encode(out);
+        self.gas_used.encode(out);
+        self.timestamp.encode(out);
+        self.extra_data.to_vec().as_slice().encode(out);
+        self.prev_randao.encode(out);
+        B64::ZERO.encode(out);
+        self.base_fee_per_gas.encode(out);
+        withdrawals_root.encode(out);
+        self.blob_gas_used.encode(out);
+        self.excess_blob_gas.encode(out);
+        parent_beacon_block_root.encode(out);
+    }
+
+    fn rlp_payload_length(
+        &self,
+        parent_beacon_block_root: B256,
+        transactions_root: B256,
+        withdrawals_root: B256,
+    ) -> usize {
+        self.parent_hash.length()
+            + EMPTY_UNCLE_ROOT_HASH.length() // ommers_hash
+            + self.fee_recipient.length()
+            + self.state_root.length()
+            + transactions_root.length()
+            + self.receipts_root.length()
+            + self.logs_bloom.length()
+            + U256::ZERO.length() // difficulty
+            + self.block_number.length()
+            + self.gas_limit.length()
+            + self.gas_used.length()
+            + self.timestamp.length()
+            + self.extra_data.to_vec().as_slice().length()
+            + self.prev_randao.length()
+            + B64::ZERO.length() // nonce
+            + self.base_fee_per_gas.length()
+            + withdrawals_root.length()
+            + self.blob_gas_used.length()
+            + self.excess_blob_gas.length()
+            + parent_beacon_block_root.length()
+    }
+}
+
+/// Calculate the Merkle Patricia Trie root hash from a list of items
+/// `(rlp(index), encoded(item))` pairs.
+pub fn calculate_transactions_root<T>(transactions: &[T]) -> B256
+where
+    T: Encodable,
+{
+    ordered_trie_root_with_encoder(transactions, |tx: &T, buf| tx.encode(buf))
+}
+/// Calculates the root hash of the withdrawals.
```

Comment:
```suggestion
}

/// Calculates the root hash of the withdrawals.
```

can you add a space here